addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -233,6 +233,17 @@ def _generate_public_time_off_timesheets(self, employees):
                     timesheet_vals_list.append(timesheet_vals)
         return self.env['account.analytic.line'].sudo().create(timesheet_vals_list)
 
+    def _reevaluate_leaves(self, time_domain_dict):
+        """
+        override method from hr_holidays to unlink the associated timesheets of the leaves
+        if the new duration is 0 and a holiday is associated.
+        """
+        if (leaves := super()._reevaluate_leaves(time_domain_dict)):
+            for leave in leaves:
+                if leave.number_of_days == 0 and leave.timesheet_ids and leave.timesheet_ids.holiday_id:
+                    leave.timesheet_ids.holiday_id = False
+                    leave.timesheet_ids.unlink()
```

Comment:
`_reevaluate_leaves` doesn't return anything, does it? :thinking:
IMO you could reevaluated anyway, since it only checks for empty leaves it shouldn't be an issue
```suggestion
        super()._reevaluate_leaves(time_domain_dict)
        for leave in self:
            if leave.number_of_days == 0 and leave.timesheet_ids and leave.timesheet_ids.holiday_id:
                leave.timesheet_ids.holiday_id = False
                leave.timesheet_ids.unlink()
```

Reply:
Thank you for the review @Bertrand2 !

I am afraid your suggested approach will not work, because the leaves are not in `self`(which is an `resource.calendar.leaves` object)
As you mentioned, `_reevaluate_leaves` originally does not return anything. The leaves are fetched based of the domain passed to `_reevaluate_leaves` [here](https://github.com/odoo-dev/odoo/blob/9448235cba42ef09b14ae2336b93e190849c274d/addons/hr_holidays/models/resource.py#L56)

My approach was to avoid making redundant fetch call by returning the changed leaves in the method, so that the override has direct access to it.
Please let me know if you have other ideas :)

Reply:
ok, makes sense then :) Not sure I'm a big fan of the return type able to vary given the context, but I don't know what are the conventions in your team.
An alternative solution could also be to make that logic upon writing the leaves; if the new number of days is 0, timesheets can be unlinked.
Last thing, I just noticed that `unlink` could be done in batch outside of the loop :ok_hand:

Reply:
Thanks for the feedback! I opted with your suggestion by making a override on write instead and reverted the changes to `_reevaludate_leaves`.